co-simulation/lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/transform/Proj4Projection.java

```diff
@@ -0,0 +1,195 @@
+
+import static java.lang.Math.toDegrees;
+import static java.lang.Math.toRadians;
+
+
+import org.eclipse.mosaic.lib.geo.CartesianPoint;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.geo.MutableCartesianPoint;
+import org.eclipse.mosaic.lib.geo.MutableGeoPoint;
+import org.eclipse.mosaic.lib.geo.MutableUtmPoint;
+import org.eclipse.mosaic.lib.geo.UtmPoint;
+import org.eclipse.mosaic.lib.geo.UtmZone;
+import org.eclipse.mosaic.lib.math.MathUtils;
+import org.eclipse.mosaic.lib.math.Vector3d;
+
+
+import org.locationtech.proj4j.CRSFactory;
+import org.locationtech.proj4j.CoordinateReferenceSystem;
+import org.locationtech.proj4j.CoordinateTransform;
+import org.locationtech.proj4j.CoordinateTransformFactory;
+import org.locationtech.proj4j.ProjCoordinate;
+
+
+
+public class Proj4Projection extends GeoProjection {
+
+    // private final String georeference;
+    private GeoPoint geoOrigin;
+    UtmZone zone;
+
+    private double x_offset;
+    private double y_offset;
+
+    private String wgs84ReferenceString;
+    private String cartesianReferenceString;
+    CoordinateReferenceSystem wgs84CRS;
+    CoordinateReferenceSystem cartesianCRS;
+
+
+    public Proj4Projection(GeoPoint origin, double x_offset, double y_offset, String georef){
+        this.geoOrigin = origin;
+        this.x_offset = x_offset;
+        this.y_offset = y_offset;
+
+        // The proj library being used currently doesn't support use of vertical keywords
+        //such as geodgrids and vunits, resulting in no vertical accuracy in the transforms calculated
+        //Isssue tracked under: https://github.com/locationtech/proj4j/issues/20
+        //Remove unaccepted parameters from georeference
+        String regex = "(\\+geoidgrids=[^\\s]+\\s?)|(\\+vunits=[^\\s]+\\s?)";
+
+        this.cartesianReferenceString = georef.replaceAll(regex, "").trim();;
+        this.wgs84ReferenceString = "EPSG:4326";
+
+        CRSFactory crsFactory = new CRSFactory();
+        this.cartesianCRS = crsFactory.createFromParameters("custom_proj", this.cartesianReferenceString);
+        this.wgs84CRS = crsFactory.createFromName(wgs84ReferenceString);
+
+    }
+
+    @Override
+    public Vector3d geographicToVector(GeoPoint geographic, Vector3d result){
+        getGeoCalculator().distanceBetween(geoOrigin, geographic, result);
+        return result;
+    }
+
+    @Override
+    public MutableGeoPoint vectorToGeographic(Vector3d vector3d, MutableGeoPoint result) {
+        getGeoCalculator().pointFromDirection(geoOrigin, vector3d, result);
+        return result;
+    }
+
+    @Override
+    public MutableCartesianPoint geographicToCartesian(GeoPoint geographic, MutableCartesianPoint result) {
+
+        ProjCoordinate sourceCoord = new ProjCoordinate(geographic.getLongitude(),geographic.getLatitude());
+        ProjCoordinate targetCoord = new ProjCoordinate();
+        CoordinateTransformFactory ctFactory = new CoordinateTransformFactory();
```

Comment:
nit: Is there a reason to initialize a new factory every time we call this function? Couldn't we init it in the constructor and just use the same factory each time?

Reply:
Same the transform below maybe?